planner/core/point_get_plan.go

```diff
@@ -1419,6 +1419,10 @@ func buildOrderedList(ctx sessionctx.Context, plan Plan, list []*ast.Assignment,
 			Col:     col,
 			ColName: plan.OutputNames()[idx].ColName,
 		}
+		defaultExpr := extractDefaultExpr(assign.Expr)
+		if defaultExpr != nil {
+			defaultExpr.Name = assign.Column
+		}
```

Comment:
Since you're modified the point get, do you need to test it?

Reply:
It was detected/tested by another unit test:
ddl/db_test.go: TestDefaultSQLFunction
"update t2 set b = default, c = default WHERE a = 2"
after my changes, so it was needed :)